include/xgboost/tree_model.h

```diff
@@ -613,6 +626,107 @@ inline void RegTree::CalculateContributions(const RegTree::FVec& feat, unsigned
   out_contribs[split_index] += leaf_value - node_value;
 }
 
+// used by _compute_weight
+inline unsigned _seq_prod(unsigned a, unsigned b) {
+  unsigned out = 1.0;
+  for (unsigned i = a; i <= b; ++i) out *= i;
+  return out;
+}
+
+inline bst_float _compute_weight(unsigned num_fixed, unsigned num_integrated) {
```

Comment:
I'll change the naming to CamelCase. As for the lookup table, you would think that, but I had one and took it out because it didn't help much. I didn't try log gamma. My guess is the reason for the lack of change with a table was that another aspect of the method was dominating the run time.

I'll profile things and see if I can get better insight into this.